docs/notebooks/2.3 Dust Soiling with Kimber.ipynb

```diff
@@ -0,0 +1,398 @@
+       "      <th></th>\n",
+       "      <th>precipitation_rate</th>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>period_mid</th>\n",
+       "      <th></th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>2024-05-01 04:30:00+00:00</th>\n",
+       "      <td>0.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2024-05-01 05:30:00+00:00</th>\n",
+       "      <td>0.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2024-05-01 06:30:00+00:00</th>\n",
+       "      <td>0.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2024-05-01 07:30:00+00:00</th>\n",
+       "      <td>0.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2024-05-01 08:30:00+00:00</th>\n",
+       "      <td>0.0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                           precipitation_rate\n",
+       "period_mid                                   \n",
+       "2024-05-01 04:30:00+00:00                 0.0\n",
+       "2024-05-01 05:30:00+00:00                 0.0\n",
+       "2024-05-01 06:30:00+00:00                 0.0\n",
+       "2024-05-01 07:30:00+00:00                 0.0\n",
+       "2024-05-01 08:30:00+00:00                 0.0"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "rainfall_data = solcast.get_solcast_historic(\n",
+    "    latitude, longitude, \n",
+    "    start=start, end=end, \n",
+    "    api_key=os.getenv('SOLCAST_API_KEY'), \n",
+    "    output_parameters=['precipitation_rate'],\n",
+    "    period=\"PT1H\"\n",
+    ")[0]\n",
+    "\n",
+    "rainfall_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e14906a4-5289-4e81-ad54-f0ac02184edc",
+   "metadata": {},
+   "source": [
+    "We create a new column `accumulated_rainfall` with the daily total rainfall, i.e. the total rainfall in the preceding 24h. This is the same that the Kimber algorithm does: when this accumulation reaches the cleaning threshold, the soiling rate returns to 0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "b6da5564-cec7-4bab-8a1c-d388a52bc584",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# the local timezone is UTC - 4 hours\n",
+    "rainfall_data.index = rainfall_data.index - pd.Timedelta(hours=4)\n",
+    "\n",
+    "rainfall_data['accumulated_rainfall'] = rainfall_data.rolling(24, closed='right').sum()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "71f018e5-2d7d-425b-b31b-fd3153d1db6e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot(time_labels: pd.DatetimeIndex, precip_rate: pd.Series, precipt_daily: pd.Series, soiling: pd.Series, threshold: int=15):\n",
```

Comment:
Line too long - break the arguments out onto single lines
'precipt_daily' should be 'precip_daily' (here and within the function)